From 5eea9fda087c2454fa42e7e708439a17af7f3def Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 13 Dec 2023 14:36:49 +0200
Subject: [PATCH] Revert "[SW-0] wait for prstn in ppboot"

This reverts commit d454e581660f6c38a7734fb051ae134cacff334b.
this WA is not required anymore

Change-Id: I19f3dbf848c28dc36f3e18a9adc2976fea709bdd
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 7ddf4bd..ca1cf2d 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -819,13 +819,6 @@
 	u32 run_req_reg = data->cpu_cfg_base + PIF_RUN_HALT_REQ_OFF;
 	u64 start;
 
-	/*
-	 * TODO: temporary WA, this polling for PRSTN should be in PIF ARC
-	 * Wait for PRSTn
-	 */
-	while (!regmap_readl(D0_PSOC_GLOBAL_CONF_REG(mmGLOBAL_CONF_PRSTN)))
-		;
-
 	/* Set ARC in running state */
 	regmap_writel(PIF_ARC_RUN_HALT_REQ_RUN_REQ_M, run_req_reg);
 
@@ -1080,6 +1073,10 @@
 	u32 link_timeout;
 	enum pif_reset_source rst_src;
 
+	/* Wait for PRSTn */
+	while (!regmap_readl(D0_PSOC_GLOBAL_CONF_REG(mmGLOBAL_CONF_PRSTN)))
+		;
+
 	/* arm timer for delays and timeouts in the pcie init sequence */
 	pif_ts_counter_arm();
 
